highlight.io/pages/blog/index.tsx

```diff
@@ -103,8 +103,18 @@ export const getStaticProps: GetStaticProps = async () => {
 	let tags = await loadTagsFromGithub(posts)
 
 	posts
-		.filter((a) => new Date().getTime() - Date.parse(a.postedAt) > 0)
 		.sort((a, b) => Date.parse(b.postedAt) - Date.parse(a.postedAt))
+		.map((a) => {
+			if (new Date().getTime() - Date.parse(a.postedAt) < 0) {
+				console.log('hiding future post', {
+					slug: a.slug,
+					postedAt: a.postedAt,
+					parsed: Date.parse(a.postedAt),
+				})
+			}
+			return a
+		})
+		.filter((a) => new Date().getTime() - Date.parse(a.postedAt) >= 0)
```

Comment:
I clicked tested in [the PR preview](https://highlight-landing-jtng3ji5q-highlight-run.vercel.app/blog?page=1) and it seems like the future posts are still being shown.

<img width="1046" alt="Screenshot 2024-04-30 at 5 15 17 PM" src="https://github.com/highlight/highlight/assets/308182/ff7a85f5-8636-45c4-94bc-ec36994db505">

Wondering if this is because we're not reassigning `posts` and just calling `filter` on it which won't transform the original array but return a new one with the filtered results?

Reply:
I pushed that change ([f1a8531](https://github.com/highlight/highlight/pull/8400/commits/f1a853142ccaec4b2f8da9696f99adcde9e5bbc0)) - hope you don't mind!

Reply:
thank you chris! :facepalm: